### PR TITLE
Remove jQuery dependency for send-to-device.js (Fixes #7673)

### LINKED ISSUE #7673: Remove jQuery dependency for send-to-device.js
## Description

The send-to-device widget is still being used in modern templates, but relies on jQuery. We should replace those function calls with native equivalents.

---

#### :yellow_heart: Success Criteria :yellow_heart:
- [x] send-to-device.js no longer relies on jQuery.

#### Tasks
- [x] Remove dependency.
- [x] Update tests where applicable.

### PR DESCRIPTION
**Pull request by vinay-deshmukh**
Since there is no corresponding 'footer > ul' HTML in the
./bedrock/base/templates/macros.html file,
the `this.$footerLinks` has also been removed.

## Description

Removed the jQuery functions and converted send-to-device.js to purely Vanilla js. I've had to update the test file as well.

## Issue / Bugzilla link

https://github.com/mozilla/bedrock/issues/7673

## Testing

I have run the karma unit tests through a local installation and they are all passing.

### CODE REVIEW

File: media/js/base/send-to-device.js [hunk 1]
```diff
@@ -9,29 +9,28 @@ if (typeof window.Mozilla === 'undefined') {
...
-        this.$input = this.$formFields.find('.send-to-device-input');
-        this.$thankyou = this.$widget.find('.thank-you');
-        this.$errorList = this.$form.find('.error-list');
-        this.$spinnerTarget = this.$form.find('.loading-spinner');
-        this.$footerLinks = this.$widget.find('footer > ul');
-        this.$sendAnotherLink = this.$form.find('.send-another');
-        this.$formHeading = this.$widget.find('.form-heading');
-        this.spinnerColor = this.$widget.data('spinnerColor') || '#000';
-        this.countries = this.$widget.data('countries');
+        this.$widget = document.getElementById(this.formId);
```
Comment (alexgibson) [inline media/js/base/send-to-device.js:23]: The prefix of adding `$` to the start of variable names is from a long gone convention from when we wanted to distinguish between native DOM objects and jQuery objects. Now that we're removing jQuery from this file, I think we're fine to remove `$` from the beggining of the variable names.

File: media/js/base/send-to-device.js [hunk 2]
```diff
@@ -58,7 +57,7 @@ if (typeof window.Mozilla === 'undefined') {
      * Initialise the form messaging and bind events.
      */
     SendToDevice.prototype.init = function() {
-        if (this.$widget.length === 1) {
+        if (this.$widget instanceof HTMLElement) {
```
Comment (alexgibson) [inline media/js/base/send-to-device.js:60]: I think we're OK to shorten this to `if (this.widget) {`?

File: media/js/base/send-to-device.js [hunk 3]
```diff
@@ -90,17 +89,19 @@ if (typeof window.Mozilla === 'undefined') {
         // just show the email messaging after 5 seconds waiting.
         this.formTimeout = setTimeout(self.updateMessaging, 5000);
 
-        $.get('/country-code.json')
-            .done(function(data) {
+        window.fetch('/country-code.json')
```
Comment (alexgibson) [inline media/js/base/send-to-device.js:92]: `fetch` isn't supported by Internet Explorer, so we may want to use the traditional `XMLHttpRequest`

File: media/js/base/send-to-device.js [hunk 4]
```diff
@@ -134,59 +135,76 @@ if (typeof window.Mozilla === 'undefined') {
      * Updates the form fields to include SMS messaging
      */
     SendToDevice.prototype.showSMS = function() {
-        var $label = this.$formFields.find('.form-input-label');
-        this.$form.addClass('sms-country');
-        $label.html($label.data('alt'));
-        this.$input.attr('placeholder', this.$input.data('alt'));
+        var $label = document.querySelector('.form-input-label');
+        this.$form.classList.add('sms-country');
+        $label.innerHTML = $label.dataset.alt;
```
Comment (alexgibson) [inline media/js/base/send-to-device.js:140]: Please use `getAttribute('data-alt')` for legacy browsers (here and below).

File: media/js/base/send-to-device.js [hunk 5]
```diff
@@ -134,59 +135,76 @@ if (typeof window.Mozilla === 'undefined') {
...
      * Binds form submission and click events
      */
     SendToDevice.prototype.bindEvents = function() {
-        this.$form.on('submit', $.proxy(this.onFormSubmit, this));
-        this.$sendAnotherLink.on('click', $.proxy(this.sendAnother, this));
+        this.eventFormSubmit = this.onFormSubmit.bind(this);
+        this.$form.addEventListener('submit', this.eventFormSubmit);
+
+        if(this.$sendAnotherLink) {
+            // Check that the Element exists, as it's not present in unit test spec file
```
Comment (alexgibson) [inline media/js/base/send-to-device.js:153]: If the link isn't present in the unit test mock, I'd perhaps favor adding it there versus adding checks in production code around it?

File: media/js/base/send-to-device.js [hunk 6]
```diff
@@ -134,59 +135,76 @@ if (typeof window.Mozilla === 'undefined') {
...
         e.preventDefault();
-        this.$input.val('');
-        this.$errorList.addClass('hidden');
-        this.$thankyou.addClass('hidden');
-        this.$formHeading.removeClass('hidden');
-        this.$formFields.removeClass('hidden');
-        this.$input.trigger('focus');
+        this.$input.value = '';
+        this.$errorList.classList.add('hidden');
+        this.$thankyou.forEach(
```
Comment (alexgibson) [inline media/js/base/send-to-device.js:178]: Please try and use `for` loops over `forEach` on NodeLists for older browsers.

File: tests/unit/spec/base/send-to-device.js [hunk 7]
```diff
@@ -184,70 +228,132 @@ describe('send-to-device.js', function() {
...
-            });
+                var pd = window.Promise.resolve(
+                    new Response(
+                        new Blob([JSON.stringify(data)])
+                    )
+                );
+                return pd;
+            };
+
+            // Specifies the delay to use in setTimeout for `expect` calls
```
Comment (alexgibson) [inline tests/unit/spec/base/send-to-device.js:243]: This feels like something that a promise might resolve, versus using a setTimeout?

Comment (vinay-deshmukh) [reply tests/unit/spec/base/send-to-device.js:243]: Using a fetch().then().then() inside the top level function necessitated the use of a setTimeout as the top level function doesn't directly return a promise.

I tried other approaches with done, and async but none of them worked as intended.

However, now that I can't use fetch, these tests will again change

**Comment by alexgibson** [comment]
Looks like this needs a rebase

**Comment by vinay-deshmukh** [comment]
Yes definitely.
I'm still in the process of doing the suggested changes.

I will rebase and re request a review after the changes are done.

**Comment by alexgibson** [comment]
@vinay-deshmukh hi just checking in. Is this something you still have time to work on? No problem if not, just let me know and we can see if another contributor would like to work on it. Thanks

**Comment by alexgibson** [comment]
Closing due to inactivity - please feel free to reopen if / when you have time. Thanks
